### ISSUE TITLE
Transport stays TCP for all subsequent requests after udp-mtu is exceeded

### ISSUE THREAD

**Issue by etamme** [drachtio/drachtio-server#296]
Tested against v0.8.22

When drachtio-server receives a request of size greater than udp-mtu, it correctly switches the transport to TCP, however it continues sending TCP for all subsequent requests, even though it sets Via headers for UDP.  To clarify this is all requests, not just requests associated with the dialog whose request that exceeded udp-mtu.

The test case is simple to replicate with sipp, create a simple offer/answer scenario where the INVITE request size is greater than udp-mtu, and then send a second offer/answer where the INVITE request size is less than udp-mtu.  Both requests will be forwarded via TCP, the first request will correctly have a Via header with a tcp transport.  The second request will be sent over TCP, but will have a Via header with a udp transport.

**Comment by davehorton** [comment]
thanks can you attach the sipp scenario file you used for testing

**Comment by etamme** [comment]
I created [sipp_test.zip](https://github.com/drachtio/drachtio-server/files/11895995/sipp_test.zip) which contains the test script, test.sh, and the associated scenario files.

**Comment by etamme** [comment]
Here is a [pcap](https://github.com/drachtio/drachtio-server/files/11896012/biginvite.zip) captured when the test script is run

**Comment by lylepratt** [comment]
We seem to also be having this issue. We noticed requests to a particular FS started using TCP for all of its requests (other FS boxes were still using UDP) and just as mentioned in this article, the VIA shows UDP. Invites to this FS that got into this mode started failing every time.

**Comment by dualtone-tony** [comment]
Been suffering this same issue, came to the same conclusion and then found this issue. Was there any progress on this @davehorton @xquanluu ?

**Comment by etamme** [comment]
This is a sofia-sip issue I believe, you can override the max udp mtu per the docs https://drachtio.org/docs/drachtio-server#udp-mtu

**Comment by dualtone-tony** [comment]
> This is a sofia-sip issue I believe, you can override the max udp mtu per the docs https://drachtio.org/docs/drachtio-server#udp-mtu

Yep now done this, but doesn't feel like a great workaround for genuine mtu size issues. Any links to the underlying sofia-sip issue? We use freeswitch and it's never suffered this with sofia?

**Comment by etamme** [comment]
"The UDP size limit of 1300 bytes is enforced by default. If limit is exceeded, TCP is tried instead. If TCP connection is refused, UDP is tried if message size is less than 64 kilobytes. Limit is adjustable via parameter NTATAG_UDP_MTU" from https://sofia-sip.sourceforge.net/refdocs/sofia_sip_conformance.html
